### DISCUSSION TITLE
Remote debugging in worker mode

### DISCUSSION THREAD

**Discussion by hazadam** [php/frankenphp#495] [Q&A] [answered]
Hello :slightly_smiling_face: 

I'm trying to remote debug my worker script with xdebug in PHPStorm. When I tell PHPStorm to break on first line, it does so, **but since the $_SERVER["SERVER_NAME"] is an empty string**, I cannot map directories to actually remote debug my application. **Is there any way to set the "SERVER_NAME" on franken startup?** I'm guessing that what triggers xdebug initially is the "dummy" request, which has a very limited set of headers.

I thought of triggering xdebug on a legit request by **setting FRANKENPHP_LOOP_MAX=1 to make the php script die soon and start again, but I'm getting sigsegv there right now.**

How do you guys go about this?

Thank you :slightly_smiling_face:

**Comment by domis86** [comment]
see: https://github.com/dunglas/frankenphp/discussions/492

**Comment by hazadam** [comment, marked as answer]
After having further [discussions](https://github.com/dunglas/frankenphp/discussions/492) and digging deeper I discovered that there is actually a very simple solution to this. FrankenPHP passes ENV variables defined under given workers as $_SERVER variables into your php scripts. Once I had the "SERVER_NAME" variable set up, PHPStorm identified it and used the associated path mapping to let me debug my worker script just like I'm used to under "normal" circumstances.

ENVs are passed like this 🔽 
```
	frankenphp {
	    num_threads 2
		worker {
		    file /app/public/index.php
		    num 1
		    env SERVER_NAME mailship.local
		}
	}
```
